File: UzaySimulasyonu(C)/src/Kisi.c
#include "Kisi.h"

// Kurucu fonksiyon
Kisi kisiOlustur(char* isim, int yas, float kalanOmur, Arac bulunduguArac) {
    Kisi this = (Kisi)malloc(sizeof(struct KISI));
    
    this->isim = strdup(isim);
    this->yas = yas;
    this->kalanOmur = kalanOmur;
    this->bulunduguArac = bulunduguArac;
    this->yasiyorMu = 1;

    this->bulunduguArac->yolcuSayisiGuncelle(this->bulunduguArac, +1); // bulunduğu aracın yolcu sayısı +1

    this->birSaatIlerle = &kisiBirSaatIlerle;
    this->yoket = &kisiYoket;

    return this;
}

// 1 saat simule eder. ömrü biterse ölür
void kisiBirSaatIlerle(const Kisi this) {
    if (this->bulunduguArac->bulunduguGezegen != NULL) {
        this->kalanOmur -= this->bulunduguArac->bulunduguGezegen->yaslanmaCarpani;
    } else {
        this->kalanOmur -= 1;
    }
    if (this->kalanOmur <= 0 && this->yasiyorMu == 1) {
        this->bulunduguArac->yolcuSayisiGuncelle(this->bulunduguArac, -1);
        this->yasiyorMu = 0;
    }
}

// Yıkıcı fonksiyon
void kisiYoket(const Kisi this) {
    if (this == NULL) return;
    free(this->isim);
    free(this);
}

File: UzaySimulasyonu(Java)/src/Sınıflar/Kisi.java
/**
*
* @author Yunus Emre İnci yunus.inci1@ogr.sakarya.edu.tr
* @since 25.04.2025
* <p>
* Kisi Sınıfı;
* Kisi nesnelerinin değişkenlerini ve saat ilerleme fonksiyonunu barındırır.
* </p>
*/

package Sınıflar;

public class Kisi {
	@SuppressWarnings("unused")
	private String isim;
	@SuppressWarnings("unused")
	private int yas;
	private int kalanOmur;
	private Arac bulunduguArac;
	
	public Kisi(String isim, int yas, int kalanOmur, Arac bulunduguArac) { // Kurucu fonksiyon
		this.isim = isim;
		this.yas = yas;
		this.kalanOmur = kalanOmur;
		this.bulunduguArac = bulunduguArac;
		
		bulunduguArac.yolcuSayisiGuncelle(+1); // Arac yolcu sayısı +1
	}
	
	public void xSaatIlerle(int x) { // x saat ilerle fonksiyonu
		if (kalanOmur <= 0) return;
		kalanOmur -=x;                             // Ömrü x saat azaldı
		if (kalanOmur <= 0) {
			bulunduguArac.yolcuSayisiGuncelle(-1); // Öldüyse arac yolcu sayısı -1
		}
	}
}
